### DISCUSSION TITLE
Fetch data from API and write result from JSON-path in column

### DISCUSSION THREAD

**Discussion by webwurm** [AntlerVC/firetable#823] [Support / Q&A] [answered]
I started with Rowy a few days ago - and I want to say: Thanks for that great tool. It is a great addition to building apps on Flutterflow. And - coming from that side - I have one question which I can't figure out:

In FlutterFlow I use a call to the Google NearbySearch-API to get the coordinates of the next metro-station after typing in a new object and its coordinates, like so:

https://maps.googleapis.com/maps/api/place/nearbysearch/json?`${query}&radius=1000&type=subway_station&location=48.202360%2C16.349060&key=[my_key]`

From the returned JSON I get the coordinates with:
`$.results[0].geometry.location` (or `$.results[0].geometry.location.lat` und `$.results[0].geometry.location.lng`)

Then I write it in a field in the objects-collection in Firebase.

To run that on the client-side on the phone is actually really not smart - it would be great to run this on the server:
1. A new row is created
2. Data from the Google-API is fetched
3. The result is written in a field/column

I thought Rowy is the right tool for that and started making a Connector-column in my Objects-table. This is how the function looks like now:

```
const connectorFn: Connector = async ({query, row, user}) => {

    const response = await fetch(
    `https://maps.googleapis.com/maps/api/place/nearbysearch/json?${query}&radius=1000&type=subway_station&location=48.202360%2C16.349060&key=[my_key]`
    );
    const { answer } = await response.json();
    return answer;
};
```
But some things I can't figure out:
* The "location" should be a query-parameter - how do I set that?
* Where do I tell it, which parts of the json to return?

Maybe somebody can point me in the right direction - I checked the docs with the FruitPicker, but I can't get this to work.
Thanks!

**Comment by shamsmosowi** [comment, marked as answer]
Hi @webwurm, if the document has geolocation data, you can access it through the row parameter in the connector function 
i've update the demo to include an example use case [here]( https://demo.rowy.io/table/geolocation#columnModal=%7B%22type%22%3A%22config%22%2C%22columnKey%22%3A%22mapsSearch%22%7D)

**Reply by webwurm** [reply]
Thank you very, very much! Awesome!

**Reply by webwurm** [reply]
Okay, I got it to work in my usecase too. But I have 2 (small) follow-up questions:

1. I don't know why the ${...} writing doesn't work for me...it took me a few hours to figure out that
`const res = await fetch("https://maps.googleapis.com/maps/api/place/nearbysearch/json?location=" + row.Koordinaten.latitude + "," + row.Koordinaten.longitude + "&type=subway_station&radius=1000&key="+mapsAPISecret)`
works, but the same with
`${row.Koordinaten.latitude},${row.Koordinaten.longitude]` and `${mapsAPISecret}` inline didn't work.

2. I want to return coordinates as field type GeoPoint. How do I convert the longitude-/latitude-strings I get back from the API into the GeoPoint-format?

Thanks a lot!

**Reply by shamsmosowi** [reply]
1. not sure why the template string isn't working, you can share an example, you can see it's working in the demo.
2. unfortunately  you can't directly store geopoint in the connector field atm, you can create a derivative column though

**Reply by webwurm** [reply]
Thanks, I will create a derivative column!
